src/metadata.rs

```diff
@@ -282,22 +282,37 @@ impl Metadata21 {
 
         let extra_metadata = cargo_toml.remaining_core_metadata();
 
-        let description: Option<String>;
-        let description_content_type: Option<String>;
+        let mut description: Option<String> = None;
+        let mut description_content_type: Option<String> = None;
         // See https://packaging.python.org/specifications/core-metadata/#description
-        if let Some(ref readme) = cargo_toml.package.readme {
+        // and https://doc.rust-lang.org/cargo/reference/manifest.html#the-readme-field
+        if cargo_toml.package.readme == Some("false".to_string()) {
+            // > You can suppress this behavior by setting this field to false
+        } else if let Some(ref readme) = cargo_toml.package.readme {
             let readme_path = manifest_path.as_ref().join(readme);
             description = Some(fs::read_to_string(&readme_path).context(format!(
-                "Failed to read readme specified in Cargo.toml, which should be at {}",
+                "Failed to read Readme specified in Cargo.toml, which should be at {}",
                 readme_path.display()
             ))?);
 
             description_content_type = extra_metadata
                 .description_content_type
                 .or_else(|| Some(path_to_content_type(&readme_path)));
         } else {
-            description = None;
-            description_content_type = None;
+            // > If no value is specified for this field, and a file named
+            // > README.md, README.txt or README exists in the package root
+            for readme_guess in ["README.md", "README.txt", "README"] {
```

Comment:
Add `README.rst`?

Reply:
Done. Cargo itself doesn't pick up .rst since that can't be parsed in the rust world (esp. on crates.io), but it makes sense for python projects